syntax_and_semantics/literals/regex.md

```diff
@@ -1,35 +1,67 @@
-# Regex
+# Regular Expressions
 
-Regular expressions are represented by the [Regex](http://crystal-lang.org/api/Regex.html) class, which is usually created with a literal:
+Regular expressions are represented by the [Regex](http://crystal-lang.org/api/Regex.html) class.
+
+A Regex is typically created with a regex literal using [PCRE](http://pcre.org/pcre.txt) syntax. It consists of a string of UTF-8 character enclosed in forward slashes (`/`):
 
 ```crystal
-foo_or_bar = /foo|bar/
-heeello    = /h(e+)llo/
-integer    = /\d+/
+/foo|bar/
+/h(e+)llo/
+/\d+/
+/ 
 ```
 
-A regular expression literal is delimited by `/` and uses [PCRE](http://pcre.org/pcre.txt) syntax.
-
-It can be followed by these modifiers:
-
-* i: ignore case (PCRE_CASELESS)
-* m: multiline (PCRE_MULTILINE)
-* x: extended (PCRE_EXTENDED)
+## Escaping
 
-For example
+Regular expressions support the same [escape sequences as String literals](./string.html).
 
 ```crystal
-r = /foo/imx
+/\// # slash
+/\\/ # backslash
+/\b/ # backspace
+/\e/ # escape
+/\f/ # form feed
+/\n/ # newline
+/\r/ # carriage return
+/\t/ # tab
+/\v/ # vertical tab
+/\NNN/ # octal ASCII character
+/\xNN/ # hexadecimal ASCII character
+/\uNNNN/ # hexadecimal unicode character
+/\u{NNNN...}/ # hexadecimal unicode character
 ```
 
-Slashes must be escaped:
+The delimiter character `/` must be escaped inside slash-delimited regular expression literals.
+Note that special characters of the PCRE syntax need to be escaped if they are intended as literal characters.
+
+## Interpolation
+
+Interpolation works in regular expression literals just as it does in [string literals](./string.html). Be aware that using this feature will cause an exception to be raised at runtime, if the resulting string results in an invalid regular expression.
+
+## Modifiers
+The closing delimiter may be followed by a number of optional modifiers to adjust the matching behaviour of the regular expression.
+
+* `i`: case-insensitive matching (`PCRE_CASELESS`):  Unicode letters in the pattern match both upper and lower case letters in the subject string.
```

Comment:
I'd rather not document the internal pcre flag names here.

Reply:
I'd argue they are useful if you want to know more details about the behaviour of the modifiers.

Reply:
I'd say you need to document it better then.